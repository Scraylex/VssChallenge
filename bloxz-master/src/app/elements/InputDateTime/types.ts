import { FormikType } from 'shared/types'

export type InputDateTimeProps = {
  formik: FormikType
  label: string
  name: string
}
