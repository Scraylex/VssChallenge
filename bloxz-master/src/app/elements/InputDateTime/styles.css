.form-control .input-date-time {
  background: var(--ion-color-background-card);
  height: 35px;
  box-shadow: 1px 3px 1px rgba(0, 0, 0, 0.25);
  border: 0;
  border-radius: 5px;
  padding: 5px 10px;
}
