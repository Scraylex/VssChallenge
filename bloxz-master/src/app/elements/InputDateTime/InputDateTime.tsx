import React from 'react'
import { IonDatetime } from '@ionic/react'
import { DatetimeChangeEventDetail } from '@ionic/core/dist/types/interface'
import { Caption } from 'app/fundamentals/Typography'
import { InputDateTimeProps } from './types'
import FormControl from '../FormControl'
import './styles.css'

const InputDateTime = ({ name, label, formik: { setFieldValue, values, touched, errors } }: InputDateTimeProps) => {
  return (
    <FormControl
      error={!!(touched[name] && errors[name])}
      errorMsg={errors[name] as string}
      data-testid="input-date-time"
    >
      <Caption data-testid="input-date-time__label">{label}</Caption>

      <IonDatetime
        date-testid="input-date-time__input"
        className="input-date-time"
        displayFormat="D MMM YYYY H:mm"
        value={values[name]}
        // Not testable because we cannot modify this ionic element with jest
        onIonChange={({ detail: { value } }: CustomEvent<DatetimeChangeEventDetail>) => {
          setFieldValue(name, value)
        }}
        max={new Date(new Date().setFullYear(new Date().getFullYear() + 10)).toISOString()}
        min={new Date(Date.now()).toISOString()}
      />
    </FormControl>
  )
}

export default InputDateTime
