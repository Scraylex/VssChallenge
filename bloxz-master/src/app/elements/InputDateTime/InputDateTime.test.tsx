import React from 'react'
import { render } from '@testing-library/react'
import { useFormik } from 'formik'
import * as yup from 'yup'
import InputDateTime from './InputDateTime'

const WrappedMUTWithValidation = () => {
  const mockFormik = useFormik({
    validationSchema: yup.object().shape({
      test: yup.date().required(),
    }),
    initialValues: { props: {} },
    onSubmit: () => {
      jest.fn()
    },
  })
  return <InputDateTime label="Test" name="test" formik={mockFormik} />
}

describe('Input DateTime', () => {
  it('should able to set datetime', async () => {
    const { asFragment } = render(<WrappedMUTWithValidation />)

    expect(asFragment).toMatchSnapshot()
  })
})
