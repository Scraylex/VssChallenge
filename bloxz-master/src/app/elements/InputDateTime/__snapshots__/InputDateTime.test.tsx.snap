// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input DateTime should able to set datetime 1`] = `[Function]`
